python/ray/autoscaler/_private/gcp/node.py

```diff
@@ -383,24 +384,43 @@ def set_labels(self,
 
         return result
 
+    def _convert_resources_to_urls(
+            self, configuration_dict: Dict[str, Any]) -> Dict[str, Any]:
+        configuration_dict = deepcopy(configuration_dict)
+        existing_machine_type = configuration_dict["machineType"]
+        if not re.search("zones/*/machineTypes/*", existing_machine_type):
```

Comment:
Is this regex correct? Shouldn't it be `. *`?

Reply:
I'm a bit iffy about using regexes for this in the first place. Maybe we can just do something with `str.split`?

Reply:
up to you about regexes, some comments/doc-strings in this function would be nice though

Reply:
I would prefer to keep a regex, but I'll loosen it `.*/machineTypes/.*`

Reply:
I would prefer to keep a regex, but I'll loosen it `.*/machineTypes/.*`